### DISCUSSION TITLE
Watch for file changes and build application

### DISCUSSION THREAD

**Discussion by gkssjovi** [vknabel/vscode-swift-development-environment#114] [Q&A]
Hello,
It is possible to preview the application, everytime when source code files are changing? 
Like a live preview.
I'm new on swift, and I don't know if this is possible on not.
txh

**Comment by vknabel** [comment]
Hi @gkssjovi. Sorry for the late response!

Sadly this is not possible to the extend you might be used from JavaScript or other scripting languages.
The easiest way to get a live preview is to use Xcode Playgrounds on macOS or on iPadOS. Though this isn't suitable for actual projects, but great for experiments.

Using VS Code, this isn't possible. Though you could set up a watcher for file changes and actually run/restart your application.
But this approach highly depends on your setup. If you need help and like to share your set up, I could have a look on it, to provide some more details.

If you are using test driven development or want to try it, set the setting `sde.swiftBuildingParams` to `test` (default: `build`). This will automatically run your tests consecutively.

**Comment by gkssjovi** [comment]
Hello @vknabel,

Thank you for your late response ;)
For the moment I switched to Flutter since it has the hard reload option. At the point I have asked the question I was not sure to use Swift or Flutter ( for cross platforms ).

Thx

**Reply by vknabel** [reply]
If the reloading was required, Flutter was a better choice. Running Swift on Android is still bleeding edge. Not sure if doing UI with Swift on Android is possible. 
I guess you decided right 👍
